app/Http/Controllers/Auth/LoginController.php

```diff
@@ -240,6 +240,16 @@ public function getTwoFactorEnroll()
      */
     public function getTwoFactorAuth()
     {
+        if (!Auth::check()) {
```

Comment:
This feels like it's doing more than what it says it's supposed to do.

It seems that _if_ this function gets called, *and* either the `two_factor_secret` is blank, or the `two_factor_enrolled` is not enabled - then we bounce you to `two-factor-enroll`.

So I'm assuming this function only gets called if 2fa is enabled or something? I'm not sure. But maybe we can either rename this or move the logic around so it doesn't feel like something else is making the decision about how to call this. Or, we can document how this works and under what circumstances it's being called in?

Reply:
> So I'm assuming this function only gets called if 2fa is enabled or something?

We have middleware that forces that redirect if necessary. 
https://github.com/snipe/snipe-it/blob/master/app/Http/Middleware/CheckForTwoFactor.php

That's there just in case you have a browser history/etc situation. You wouldn't normally find yourself on that page if the the middleware conditions weren't met.